langchain/src/chat_models/googlevertexai/common.ts

```diff
@@ -214,19 +228,46 @@ export class BaseChatGoogleVertexAI<AuthOptions>
     return [];
   }
 
-  // TODO: Add streaming support
+  async *_streamResponseChunks(
+    _messages: BaseMessage[],
+    _options: this["ParsedCallOptions"],
+    _runManager?: CallbackManagerForLLMRun
+  ): AsyncGenerator<ChatGenerationChunk> {
+    // Make the call as a streaming request
+    const instance: GoogleVertexAIChatInstance = this.createInstance(_messages);
+    const parameters = this.formatParameters();
+    const result = await this.streamedConnection.request(
+      [instance],
+      parameters,
+      _options
+    );
+
+    // Get the streaming parser of the response
+    const stream = result.data as GoogleVertexAIStream;
+
+    // Loop until the end of the stream
+    // During the loop, yield each time we get a chunk from the streaming parser
+    // that is either available or added to the queue
+    while (!stream.streamDone) {
```

Comment:
Feels like this should be an iterator?

Reply:
I had enough trouble trying to wrap my head around implementing the async iterator correctly. I don't think I wanted to make multiple, nested, async iterators.